src/main/java/org/broadinstitute/hellbender/engine/filters/ReadFilterLibrary.java

```diff
@@ -162,26 +193,33 @@ private ReadFilterLibrary(){ /*no instance*/ }
             return read.isUnmapped() ||
                     read.getLength() == Cigar.getReadLength(read.getCigarElements());}}
 
+    /** Keep only reads that are second of pair. */
     @DocumentedFeature(groupName=HelpConstants.DOC_CAT_READFILTERS, groupSummary=HelpConstants.DOC_CAT_READFILTERS_SUMMARY)
     public static class SecondOfPairReadFilter extends ReadFilter {
         private static final long serialVersionUID = 1L;
         @Override public boolean test (final GATKRead read) {
             return read.isSecondOfPair();}}
 
+    /** Keep reads with sequenced bases. */
     @DocumentedFeature(groupName=HelpConstants.DOC_CAT_READFILTERS, groupSummary=HelpConstants.DOC_CAT_READFILTERS_SUMMARY)
     public static class SeqIsStoredReadFilter extends ReadFilter {
         private static final long serialVersionUID = 1L;
         @Override public boolean test(final GATKRead read){
             return read.getLength() > 0;}}
 
+    /** Keep reads with a valid alignment start (larger than 0, or unmapped).*/
     @DocumentedFeature(groupName=HelpConstants.DOC_CAT_READFILTERS, groupSummary=HelpConstants.DOC_CAT_READFILTERS_SUMMARY)
     public static class ValidAlignmentStartReadFilter extends ReadFilter {
         private static final long serialVersionUID = 1L;
         @Override public boolean test(final GATKRead read){
             return read.isUnmapped() || read.getStart() > 0;}}
 
-    // Alignment doesn't align to a negative number of bases in the reference.
-    //Note: to match gatk3 we must keep reads that align to zero bases in the reference.
+    /**
+     * Keep reads where the read end corresponds to a proper alignment -- that is, the read ends after the start
+     * (non-negative number of bases in the reference).
+     *
+     * <p>Note: keep also unmapped reads (align to zero bases in the reference).
+     */
```

Comment:
This filter is a bit counter-intuitive after having learned that FragmentLengthReadFilter.java uses absolute length values. So this means pairs with atypical arrangements, e.g. two reads pointing outwards instead of inwards, will be filtered?

Can we include how this is calculated:
```
start = 1-based inclusive leftmost position of the clipped sequence (0 if no position)
end = 1-based inclusive rightmost position of the clipped sequence (0 if unmapped)

end - start + 1 >=0
```